eng/scripts/vcpkg.ps1

```diff
@@ -0,0 +1,67 @@
+[CmdletBinding()]
+Param (
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $Ref = 'master',
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Dependencies,
+
+    [Parameter()]
+    [string] $TargetPath ="$env:TEMP/$([guid]::NewGuid())",
+
+    [Parameter()]
+    [switch] $CI = ($null -ne $env:SYSTEM_TEAMPROJECTID)
+)
+
+function OutputWarning {
+    param([string] $Output)
+
+    if ($CI) {
+        Write-Host "##vso[task.logissue type=warning]$Output"
+    } else {
+        Write-Warning $Output
+    }
+}
+
+function SetEnvironmentVariable {
+    param(
+        [string] $Name,
+        [string] $Value
+    )
+
+    if ($CI) {
+        Write-Host "##vso[task.setvariable variable=$Name;]$($Value)"
+    } else {
+        Write-Verbose "Setting local environment variable: $Name = ***"
+        Set-Item -Path "env:$Name" -Value $Value
+    }
+}
+
+
+New-Item -ItemType Directory -Path $TargetPath -Force
+pushd $TargetPath
+
+git clone https://github.com/Microsoft/vcpkg .
+git checkout (Get-Content "$PSScriptRoot/../vcpkg-ref.txt")
+
+if ($IsWindows) {
+    .\bootstrap-vcpkg.bat
+    .\vcpkg.exe install $Dependencies.Split(' ')
+} else {
+    ./bootstrap-vcpkg.sh
+    ./vcpkg install $Dependencies.Split(' ')
+}
+
+
+
+SetEnvironmentVariable -Name Path -Value "$TargetPath;$env:PATH"
+SetEnvironmentVariable -Name VCPKG_INSTALLATION_ROOT -Value $TargetPath
```

Comment:
Isn't this resetting the variable you already set up before calling this script?

Reply:
I suggest we depend on the script for setting them instead of the inline pwsh in the yml file.